src/Symfony/Component/Translation/MessageSelector.php

```diff
@@ -47,11 +47,11 @@ class MessageSelector
      */
     public function choose($message, $number, $locale)
     {
-        $parts = explode('|', $message);
+        $parts = preg_split('/(?<![^\\\\]\\\\)\|/', $message);
         $explicitRules = array();
         $standardRules = array();
         foreach ($parts as $part) {
-            $part = trim($part);
+            $part = trim(str_replace(['\\\\', '\|'], ['\\', '|'], $part));
```

Comment:
By the way, you shouldn't use `str_replace()` here, because it makes replace 2 times, i.e. string like "`\\|`" would be "`|`" which is wrong. `strtr()` suits for this case better.